astropy/coordinates/tests/test_frames.py

```diff
@@ -1473,3 +1492,44 @@ def test_galactocentric_references(reset_galactocentric_defaults):
                 assert k not in galcen_custom.frame_attribute_references
             else:
                 assert k in galcen_custom.frame_attribute_references
+
+
+def test_galactocentric_registry():
+    """Test registerration in galactocentric_frame_defaults"""
+    from astropy.coordinates import (ICRS,
+                                     galactocentric_frame_defaults)
+
+    # register in a new set of parameters
+    galactocentric_frame_defaults.register(
+        name="test",
+        value={
+            "galcen_coord": ICRS(ra=266.0 * u.degree, dec=-29 * u.degree),
+            "galcen_distance": 8.1 * u.kpc,
+            "galcen_v_sun": r.CartesianDifferential(
+                [12.9, 245.6, 7.8] * u.km / u.s
+            ),
+            "z_sun": 21.0 * u.pc,
+            "roll": 0 * u.deg},
+        references=None
+    )
+
+    # test registration
+    # using try/finally s.t. state is reset even if tests fail
+    try:
+        assert "test" in galactocentric_frame_defaults._registry
+        with galactocentric_frame_defaults.set("test"):
+            assert set(galactocentric_frame_defaults._state.keys()) == set()
+    except AssertionError as e:  # oops, something unexpected
```

Comment:
No need to do any `except` - the `AssertionError` will still be captured by pytest.

Reply:
Good to know.